nntrainer/layers/layer_node.h

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	graph_node.h
+ * @date	1 April 2021
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is the graph node interface for c++ API
+ */
+
+#ifndef __LAYER_NODE_H__
+#define __LAYER_NODE_H__
+
+#include <graph_node.h>
+#include <layer.h>
+#include <layer_internal.h>
+
+namespace nntrainer {
+
+/**
+ * @class   LayerNode class
+ * @brief   layer node class for the graph
+ */
+class LayerNode : public ml::train::Layer, public GraphNode {
```

Comment:
I like current status, is GraphNode gonna be type T?

Reply:
It was `type T` but not, it is not needed anymore. It was redundant and has been removed.